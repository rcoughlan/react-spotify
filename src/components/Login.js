import React from 'react';
import {
    Button,
    Col,
    Row
} from 'react-bootstrap';

const Login = () => {
    return (
        <Row>
            <Col></Col>
            <Col>
                <Button className="login-button" variant="success" size="lg" onClick={() => window.location = 'http://localhost:8888/login'}>Sign Into Spotify</Button>
            </Col>
            <Col></Col>
        </Row>
    );
}

export default Login;