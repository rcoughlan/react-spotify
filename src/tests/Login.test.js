import Login from '../components/Login';
import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Artist Component', () => { 

    it('matches the snapshot', () => {
        const tree = renderer.create(<Login />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders a button', () => {
        const wrapper = shallow(<Login />);
        const button = wrapper.find('.login-button');
        expect(button).toHaveLength(1);
    });
    
    
    it('has an onclick function that changes only port not host', () => {
        const wrapper = mount(<Login />);
        const button = wrapper.find('button.login-button');
        expect(button).toHaveLength(1);
        expect(window.location.href).toBe('http://localhost/');
        button.simulate('click');
        expect(window.location.href).toBe('http://localhost/');
    });
});