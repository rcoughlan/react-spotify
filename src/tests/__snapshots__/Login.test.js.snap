// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist Component matches the snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col"
  />
  <div
    className="col"
  >
    <button
      className="login-button btn btn-success btn-lg"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Sign Into Spotify
    </button>
  </div>
  <div
    className="col"
  />
</div>
`;
